### ISSUE TITLE
only 2 domains being blocked

### ISSUE THREAD

**Issue by gotiniens** [diginc/docker-pi-hole#60]
After the issue I reported yesterday, and you fixed rather quickly. I wanted to use the new container version. So i did the standard procedure (pull, stop, rm, run). After this I saw in the admin interface that only 2 domains were being blocked. So I tried updating the blocklist with the following command:

> docker exec pihole pihole updateGravity

I got the following output:

> ::: Aggregating list of domains...cat: can't open '/etc/pihole/list.0.raw.githubusercontent.com.domains': No such file or directory
> cat: can't open '/etc/pihole/list.1.mirror1.malwaredomains.com.domains': No such file or directory
> cat: can't open '/etc/pihole/list.2.sysctl.org.domains': No such file or directory
> cat: can't open '/etc/pihole/list.3.zeustracker.abuse.ch.domains': No such file or directory
> cat: can't open '/etc/pihole/list.4.s3.amazonaws.com.domains': No such file or directory
> cat: can't open '/etc/pihole/list.5.s3.amazonaws.com.domains': No such file or directory
> cat: can't open '/etc/pihole/list.6.hosts-file.net.domains': No such file or directory
> cat: can't open '/etc/pihole/list.7.raw.githubusercontent.com.domains': No such file or directory
> cat: can't open '/etc/pihole/list.8.ransomwaretracker.abuse.ch.domains': No such file or directory
>  done!

I guess the no such file errors are the reason only 2 domains are blocked

**Comment by diginc** [comment]
Strange, do you have a volume mount for /etc/pihole that might be causing permission problems?

what happens if you run a plain startup command without anything fancy?

```
docker pull diginc/pi-hole:debian; docker run -it --rm -e DNS1=4.2.2.2 -e DNS2=4.2.2.2 -e ServerIP=0.0.0.0 diginc/pi-hole:debian | grep gravity
debian: Pulling from diginc/pi-hole
Digest: sha256:ae2a35bb3952812ddb68600cb9a4835b0f4f8e08529241554869a91bc98ea637
Status: Image is up to date for diginc/pi-hole:debian
::: 128697 domains being pulled in by gravity...
Oct  9 16:47:05 dnsmasq[252]: read /etc/pihole/gravity.list - 104560 addresses

# ctrl+c to exit

docker pull diginc/pi-hole:alpine; docker run -it --rm -e DNS1=4.2.2.2 -e DNS2=4.2.2.2 -e ServerIP=0.0.0.0 diginc/pi-hole:alpine | grep gravity            
alpine: Pulling from diginc/pi-hole
Digest: sha256:81295572f5fe1f66eb3066f5049042a9e7757b75cace0ff696ef7a6035d567e6
Status: Downloaded newer image for diginc/pi-hole:alpine
::: 128697 domains being pulled in by gravity...
Oct  9 16:47:34 dnsmasq[226]: read /etc/pihole/gravity.list - 104560 addresses

# ctrl+c to exit
```

Maybe restart docker service in case there is a networking hiccup inside containers preventing downloading of the lists.

**Comment by gotiniens** [comment]
I can't test any further at this very moment, with your sugestion I got the same errors. Couldn't restart the docker service yet, But this is my only image running in docker.

I suspect that downloading the domainlists fails because the DNS server of the host is down while downloading them. Because the DNS server is downloading the domainlists and has not yet started. ;)

Maybe it is not smart to have the pi-hole as the only DNS server of the host. Maybe the container should have a backup DNS server while downloading. I have not decided yet.

**Comment by diginc** [comment]
That's a good point that may be worth warning others about either here or in the main pi-hole docs.  I'll have to re-examine the official docs and see if a backup DNS server is recommended or not and add it to my docs if not.  In general having your docker daemon rely on anything inside of docker it's self is a 'chicken before egg' situation that should be avoided.

I have a backup DNS server that is not my pihole so I don't think I've ran into /start.sh failing to run because DNSMasq is not running yet.  The backup DNS in my router helps keep my wife's internet running when I'm doing pi-hole testing/maintenance on my main docker container so I find if helpful for preventing outages in.  It makes it so my network isn't dependent on my docker+raid home server that is by no means 24/7/365 (uptime has been flawless but I still treat it as if it could die at any time).

This maybe fixed if I just started dnsmasq before gravity.sh runs to pull down the lists and reload the config after the lists are downloaded...it seems like a patch supporting a bad practice though.  I'll have to think about it some more.

if you don't want to add a backup at the router level then your docker daemon can have DNS manually configured with the `--dns` flag https://docs.docker.com/engine/reference/commandline/dockerd/
